sympy/solvers/ode/systems.py

```diff
@@ -0,0 +1,375 @@
+    they are non-linear.
+
+    To check if the equations are constant coefficient, we need to check if all the terms in
+    A obtained above are constant or not.
+
+    To check if the equations are homogeneous or not, we need to check if b is a zero matrix
+    or not.
+
+    Parameters
+    ==========
+
+    eqs: List
+        List of ODEs
+    funcs: List
+        List of dependent variables
+    t: Symbol
+        Independent variable of the equations in eqs
+
+    Returns
+    =======
+
+    match = {
+        'no_of_equation': len(eqs),
+        'eq': eqs,
+        'func': funcs,
+        'order': order,
+        'is_linear': is_linear,
+        'is_constant': is_constant,
+        'is_homogeneous': is_homogeneous,
+    }
+
+    Dict or None
+        Dict with values for keys:
+            1. no_of_equation: Number of equations
+            2. eq: The set of equations
+            3. func: List of dependent variables
+            4. order: A dictionary that gives the order of the
+                      dependent variable in eqs
+            5. is_linear: Boolean value indicating if the set of
+                          equations are linear or not.
+            6. is_constant: Boolean value indicating if the set of
+                          equations have constant coefficients or not.
+            7. is_homogeneous: Boolean value indicating if the set of
+                          equations are homogeneous or not.
+        This Dict is the answer returned if the eqs are linear and constant
+        coefficient. Otherwise, None is returned.
+    """
+    from sympy.solvers.solveset import linear_eq_to_matrix
+    from sympy.solvers.solvers import solve
+
+    # Error for i == 0 can be added but isn't for now
+
+    # Removing the duplicates from the list of funcs
+    # meanwhile maintaining the order. This is done
+    # since the line in classify_sysode: list(set(funcs)
+    # cause some test cases to fail when gives different
+    # results in different versions of Python.
+    funcs = list(uniq(funcs))
+
+    # Check for len(funcs) == len(eqs)
+    if len(funcs) != len(eqs):
+        raise ValueError("Number of functions given is not equal to the number of equations %s" % funcs)
+
+    # ValueError when functions have more than one arguments
+    for func in funcs:
+        if len(func.args) != 1:
+            raise ValueError("dsolve() and classify_sysode() work with "
+            "functions of one variable only, not %s" % func)
+
+    # Getting the func_dict and order using the helper
+    # function
+    order = _get_func_order(eqs, funcs)
+
+    # Not adding the check if the len(func.args) for
+    # every func in funcs is 1
+
+    rep = {func.diff(t, n): Dummy() for func in funcs for n in range(order[func] + 1)}
+    eqs_sub = [eq.subs(rep) for eq in eqs]
+
+    # Linearity check
+    try:
+        A, b = linear_eq_to_matrix(eqs_sub, rep.values())
+
+    except ValueError:
+        return None
```

Comment:
I think it would be good to refactor this so that there is an ODE analogue of `linear_eq_to_matrix` that users can use to go from a system of ODEs to a matrix and vector. That can be used here

Reply:
Ok I will refactor this function in separate commit so that there is no confusion in future when seeing the changes.